### ISSUE TITLE
Unclear functionality of 'repeat' and 'shuffle' buttons

### ISSUE THREAD

**Issue by aminb** [vanilla-music/vanilla#339]
We've had similar issues in the past, and it's unclear for new users what various states of each button mean, and a lot of people don't even know they can long-press the buttons to get a list of the states in a dialog.

The least we can do is to display a tooltip with the caption for each state change.

**Comment by adrian-bl** [comment]
Maybe we should have a tutorial or a nice help page.

But that said: We already *do* have tooltips: Users just need to long-press on the button.
That's actually a standard android feature, but many people are missing this ;-)

**Comment by un-logic** [comment]
Tutorial is a nice thing to place it on [vanillamusic.io](http://vanillamusic.io) if one be written someday, but a small inviting message saying something like: 

> Hello! And welcome to Vanilla Music - a player that is heavily utilizing long presses in its own interface.

> Seriously if you ever feel confused - simply long press any disturbing object, be it a button, list item or text label, result will be immediate.

Just showing it at first start of application should be enough for everyone :)

**Comment by aminb** [comment]
A tutorial on the website is definitely nice to have, but I believe it's best if we use something like [PaoloRotolo/AppIntro](https://github.com/PaoloRotolo/AppIntro) to display a couple of nice and short messages/images/tutorials on the first run to make sure the user sees them.

**Comment by Allanitomwesh** [comment]
Holy crap I had no idea press and hold did that.
Small bug on this particular feature,if you press and hold shuffle (and get the popup) choosing continuous shuffle doesn't turn it on in settings.

**Comment by adrian-bl** [comment]
This is not a bug: the setting option just sets the default to use while cycling through the options.

**Comment by Allanitomwesh** [comment]
Hmmn I see. A suggestion then, shuffle and continuous shuffle have the same logo,perhaps give continuous shuffle its own?

**Comment by mindforger** [comment]
as an idea, the shuffle symbol with the 4 ends bended together to some degree to symbolize the infinite symbol :)

**Comment by adrian-bl** [comment]
I'm actually thinking to remove the continous shuffle option as it is a constant source of confusion and adds very little value

**Comment by Allanitomwesh** [comment]
Make it default shuffle action

**Comment by yaymeh** [comment]
I'm still not entirely sure how "shuffle + random" are supposed to work and how people are using them. At the moment shuffle doesn't seem to work at all (for items that are already in queue) and I can't tell if that's #562 or if I'm just doing it wrong. 

As far as I can see: if the right button was replaced by a simple "shuffle current queue now" (on demand action button), that should allow for all usage cases and be a lot less confusing... or is anyone doing something with the current "left button" that wouldn't be possible with just "shuffle queue now" any more? Since albums, playlists etc can easily be added to the queue, that should take care of just about everyone's shuffling needs?

( Not sure if "repeat current track" or "stop after current track" are needed either - is anyone using those / why not just queue only one track if you really want it to run "all day long"? )

**Comment by breversa** [comment]
Agreed with that one : I actually stopped trying to understand and/or
remembre how both those buttons work. ^^
All I need is to play my whole library (stored in ONE directory tree in my
SD card, BTW. No need to scan the whole partition !) in random order. :-)

It's cool to have other options though, but they sometimes are more
confusing than really useful.

2017-03-12 22:19 GMT+01:00 yaymeh <notifications@github.com>:

**Comment by adrian-bl** [comment]
The button on the right controls how the queue advances while the button on the left changes how items get enququed.

Random mode (dice on the right side) adds a random track shortly before the queue would run out of songs.

Shuffle mode (arrows on the left side) instricts vanilla to shuffle songs/albums while they get added to the queue.

**Comment by breversa** [comment]
This explanation is useful.
Why not add it to an in-app help/usage page ?

**Comment by adrian-bl** [comment]
We probably really should add some documentation to vanillamusic.io, but i am a terrible doc writer.

If someone would like to pick this up: go ahead! :-)

**Comment by un-logic** [comment]
I am a terrible doc writer too, but I'll give it a try. In exchange I would like to obtain a nightly build without translations to ease my mental labor.

**Comment by un-logic** [comment]
Well it was a long time and still a work in progress, but here's a preview. Please don't mind the "jokes" they're for fun only (kept me awake and sane for all the time, I'm not a good comedian, I know) and will be exterminated after draft state will be passed. So watch out, it's 6 pages long already:
[Vanilla Music User Guide DRAFT](https://yadi.sk/i/DzkQUwHj3Ktiup).

**Comment by kirbyfan64** [comment]
>It’s clean and simple, meaning that you’ll see as few icons as possible, probably because Vanilla Music’s community members are really humbled when it comes to their ‘mad skillz’ with Inkscape.

Ouch...

One thing I should note is that a lot of people believe things on sight. You might want to watch some of the more easily misinterpreted jokes (like this one), because, even though *we* know you had good intentions, some idiot will inevitably take this out of context in some stupid way.

**Comment by andiandi13** [comment]
I'd like to help for that documentation by starting to write solutions to issues that I've encountered (easier to explain precisely).

I think we should open an "Issue" here, where each person would post a part of the documentation related to a question / a tip etc..

@adrian-bl sounds good ?

**Comment by adrian-bl** [comment]
Sgtm, or we could enable githubs wiki function and give some people direct wiki-only-edit access (if thats possible, in the worst case we could create a new repo for the wiki)

**Comment by andiandi13** [comment]
@adrian-bl Yes it'd be cool like that. Please let us know :)

**Comment by adrian-bl** [comment]
Not sure if i got the permissions right, but i added @andiandi13 as a collaborator for https://github.com/vanilla-music/vanilla-music.github.io/wiki

So you should now have (in theory) write permission to the wiki there after accepting the invitation.

**Comment by andiandi13** [comment]
@adrian-bl Thank you, I can create new pages.

So, for the presentation, do you want 1 page = 1 theme ? 

Because it can be pages concerning a global "help" like "Playlist management", but also "questions" that would fit in an FAQ like "Why my non indexed files have no tags ?" and the reply below.

**Comment by adrian-bl** [comment]
Hmm.. Not sure: but 1 theme per page and a global FAQ wouldnt sound like a bad idea to me

**Comment by andiandi13** [comment]
That's right, but if I create an FAQ page, can anybody who have write permission add their own question inside it ?

**Comment by adrian-bl** [comment]
Btw: nice work so far with the FAQ!

**Comment by andiandi13** [comment]
Perfect then, so to all wiki users : feel free to correct my sentences of they are mistake for English isn't my mother tongue.

@adrian-bl thanks :)

**Comment by andiandi13** [comment]
@adrian-bl I was thinking of a tricks/tips page with text AND images.

Would it be a problem to add pictures if the wiki happens to be moved somewhere else (e.g inside the app, or on the website etc.) ?

**Comment by adrian-bl** [comment]
No, that wouldn't be a big issue (assuming the pictures are hosted on github)

**Comment by andiandi13** [comment]
Yes they are.
I already created the page, which I called "User guide" (explanations of settings, tricks etc..), separate from the FAQ.

**Comment by adrian-bl** [comment]
Btw: I commited 34ad6a24b4f6bc424e54b71144faa5838d33f4ca which adds a 'Help' section to the menu which links to the wiki.

Unfortunately, Github wikis don't seem to work that well on mobile, which surprises me.
Not sure if there is anything we can do, like selecting a different theme - but in the worst case, we could import the github wiki into 'vanillamusic.io' and have a script create more mobile-friendly HTML.

**Comment by andiandi13** [comment]
> Unfortunately, Github wikis don't seem to work that well on mobile

Why is the wiki always going to desktop mode in my browsers ? I mean I never had access to the wiki in mobile even if github is it's mobile version.

> we could import the github wiki into 'vanillamusic.io' and have a script create more mobile-friendly HTML.

It sounds good but if we do that, but will the wiki be automatically updated in vanillamusic.io or we have to import the changes ?

**Comment by adrian-bl** [comment]
It seems that github simply failed to create a mobile/responsive version of the wiki, so it always looks like a desktop site.

About the update: yes, we could automatically sync the contents. Just need some time to write a script to do this.

**Comment by andiandi13** [comment]
> It seems that github simply failed to create a mobile/responsive version of the wiki

Ah OK, I thought it was intentional.

>About the update: yes, we could automatically sync the contents. Just need some time to write a script to do this.

Cool :)
